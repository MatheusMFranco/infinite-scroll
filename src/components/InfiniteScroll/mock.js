export default class {
  constructor() {
    this.root = null;
    this.rootMargin = '';
    this.thresholds = [];
  }

  disconnect() {}

  observe() {}

  takeRecords() {
    return [];
  }

  unobserve() {}
}
