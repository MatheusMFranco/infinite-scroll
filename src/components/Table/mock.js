export default {
  metadata: { totalItems: 1000000, perPage: 100, page: 1 },
  data: [
    {
      id: 'e906859036f847e68e58a577da9312a2',
      absoluteIndex: 0,
      name: 'item-0',
    },
    {
      id: '761ff52b8e6dd373fdf291a1a70df20c',
      absoluteIndex: 1,
      name: 'item-1',
    },
    {
      id: '38d3f385ea8d8bb2dcfc759ac85af6ef',
      absoluteIndex: 2,
      name: 'item-2',
    },
    {
      id: '65910a0fd76f69a08e6faa142384f327',
      absoluteIndex: 3,
      name: 'item-3',
    },
    {
      id: '05e4075a3f97ed7cb0bd40329caa2423',
      absoluteIndex: 4,
      name: 'item-4',
    },
    {
      id: 'ad36abff56ce76a479dda134a3744c3b',
      absoluteIndex: 5,
      name: 'item-5',
    },
    {
      id: '7421b1b797e5710c7ee3726a1f6e6a8b',
      absoluteIndex: 6,
      name: 'item-6',
    },
    {
      id: '3a4ef5b78524c0580352bcdc969922cd',
      absoluteIndex: 7,
      name: 'item-7',
    },
    {
      id: '226337775a640fd25fbd2f0ca9b4fa7e',
      absoluteIndex: 8,
      name: 'item-8',
    },
    {
      id: 'd1b07b282926f1293feb7c4af47a7c95',
      absoluteIndex: 9,
      name: 'item-9',
    },
    {
      id: '394f3c2800035af92506ec7c5d7f755c',
      absoluteIndex: 10,
      name: 'item-10',
    },
    {
      id: 'ab42d325bbd870cfcd706d73e6c9946f',
      absoluteIndex: 11,
      name: 'item-11',
    },
    {
      id: '6d1f0d6d154c71959b3015e5f5165fc1',
      absoluteIndex: 12,
      name: 'item-12',
    },
    {
      id: 'a51f63e7d81b293c6f66f6eaa1415838',
      absoluteIndex: 13,
      name: 'item-13',
    },
    {
      id: 'c8b9b9a564aef7e9874edde58a44abfa',
      absoluteIndex: 14,
      name: 'item-14',
    },
    {
      id: 'b43e9e67163ba6beddb37ef9e77ea7cb',
      absoluteIndex: 15,
      name: 'item-15',
    },
    {
      id: '941cf308e6b24640195900263dbea458',
      absoluteIndex: 16,
      name: 'item-16',
    },
    {
      id: '7faf4030623085f256bb139c308917bc',
      absoluteIndex: 17,
      name: 'item-17',
    },
    {
      id: '250c0eeeb64242dfaf6692140b6fb76f',
      absoluteIndex: 18,
      name: 'item-18',
    },
    {
      id: 'e3aca5b7a1478bd4f63caf03251a1fd2',
      absoluteIndex: 19,
      name: 'item-19',
    },
    {
      id: '5cdbd9d6a64bd1cfc2cd65fb14ed8efa',
      absoluteIndex: 20,
      name: 'item-20',
    },
    {
      id: '535aceab0143f7e1cd629caedef3e0ea',
      absoluteIndex: 21,
      name: 'item-21',
    },
    {
      id: 'b3b4475b7f1a4bba743c0ca229c89b90',
      absoluteIndex: 22,
      name: 'item-22',
    },
    {
      id: '38d555cee07a3207d99328f7fae8e046',
      absoluteIndex: 23,
      name: 'item-23',
    },
    {
      id: '1092543d121a53714823a50491ca7ebc',
      absoluteIndex: 24,
      name: 'item-24',
    },
    {
      id: 'aa7f9f8332803e2f3d1cd5df46cd70b1',
      absoluteIndex: 25,
      name: 'item-25',
    },
    {
      id: '2fc098759adeb42aeffc0d0f37cc5395',
      absoluteIndex: 26,
      name: 'item-26',
    },
    {
      id: 'c98ab7bd4bfe916707c347313722a854',
      absoluteIndex: 27,
      name: 'item-27',
    },
    {
      id: '266a3c68d87071a9b993456df1beb9ce',
      absoluteIndex: 28,
      name: 'item-28',
    },
    {
      id: 'dd7c8cf404981c52e644bdb4f806c1d2',
      absoluteIndex: 29,
      name: 'item-29',
    },
    {
      id: 'e865fb4953f98803b42dc8a62e7709d7',
      absoluteIndex: 30,
      name: 'item-30',
    },
    {
      id: '75c54c54313aaec82be42bf048db9ae4',
      absoluteIndex: 31,
      name: 'item-31',
    },
    {
      id: 'f89f7fe2248c07d990eb0cf2ab31162f',
      absoluteIndex: 32,
      name: 'item-32',
    },
    {
      id: '85aa8182f24277c397609e0ef4933958',
      absoluteIndex: 33,
      name: 'item-33',
    },
    {
      id: '19c03b01a6bd64b1d1761b5defbff882',
      absoluteIndex: 34,
      name: 'item-34',
    },
    {
      id: '05d48faa4c919b28fe13b4af47250291',
      absoluteIndex: 35,
      name: 'item-35',
    },
    {
      id: '347fd3e817e62c79efd5da1d6e2f821b',
      absoluteIndex: 36,
      name: 'item-36',
    },
    {
      id: '01d1df7d5f6dfdbc39b85a9455231f6c',
      absoluteIndex: 37,
      name: 'item-37',
    },
    {
      id: '65ecc7df7d65b07e6dfd84dbbd9a0d17',
      absoluteIndex: 38,
      name: 'item-38',
    },
    {
      id: 'd11b33370f73ca05eb6fd1e313531062',
      absoluteIndex: 39,
      name: 'item-39',
    },
    {
      id: '9f2447525a856fce30ba34da2438af27',
      absoluteIndex: 40,
      name: 'item-40',
    },
    {
      id: '7c22bdd5f22d18737cef995129bc113c',
      absoluteIndex: 41,
      name: 'item-41',
    },
    {
      id: '0fc0f07e09a7d396d92ee9c6875153f7',
      absoluteIndex: 42,
      name: 'item-42',
    },
    {
      id: 'd55cc93e761e14bcbb13fdc7cffad88b',
      absoluteIndex: 43,
      name: 'item-43',
    },
    {
      id: 'a6c02ec95155ed66a99d7fb1760a1330',
      absoluteIndex: 44,
      name: 'item-44',
    },
    {
      id: 'a9d398a9e2477797f0fb989a8de7d4c2',
      absoluteIndex: 45,
      name: 'item-45',
    },
    {
      id: '7cfd60059b7c629f6a1359a5fcea4d33',
      absoluteIndex: 46,
      name: 'item-46',
    },
    {
      id: '51eef11c0a3ccbe29d4ab0a7f53dad9b',
      absoluteIndex: 47,
      name: 'item-47',
    },
    {
      id: 'c6b29df3b23434b0c1ff7de887bfb230',
      absoluteIndex: 48,
      name: 'item-48',
    },
    {
      id: '2b4d92ee58d1239a140a47f6a7f75de8',
      absoluteIndex: 49,
      name: 'item-49',
    },
    {
      id: '7bdb7ab6e7f3d4f676f7e5ea920e9d70',
      absoluteIndex: 50,
      name: 'item-50',
    },
    {
      id: 'db178c56fd988d6fa32bca451fcf6d90',
      absoluteIndex: 51,
      name: 'item-51',
    },
    {
      id: 'b9eb4e6974ae35e9050a7536f4b296b1',
      absoluteIndex: 52,
      name: 'item-52',
    },
    {
      id: '511dc2dc8b5896b17da20f8ce3ba9b97',
      absoluteIndex: 53,
      name: 'item-53',
    },
    {
      id: '21ba555fdf6ea2e64ee1c887791a4e0e',
      absoluteIndex: 54,
      name: 'item-54',
    },
    {
      id: 'aec377296dce715b3e5b6a33223a35f0',
      absoluteIndex: 55,
      name: 'item-55',
    },
    {
      id: 'e06e5fec86765f4618bd9bb7d36d7523',
      absoluteIndex: 56,
      name: 'item-56',
    },
    {
      id: 'e2d4c9dc1ec7539f3127dfc57a0adfdf',
      absoluteIndex: 57,
      name: 'item-57',
    },
    {
      id: 'a7f046199534000b4054636e15052f6c',
      absoluteIndex: 58,
      name: 'item-58',
    },
    {
      id: '31f06570355be090b10b6a3451676e76',
      absoluteIndex: 59,
      name: 'item-59',
    },
    {
      id: '5393c2d06bef23521c861492cbb353ae',
      absoluteIndex: 60,
      name: 'item-60',
    },
    {
      id: 'e32c707f9f1d7d8764a9879f40119b4e',
      absoluteIndex: 61,
      name: 'item-61',
    },
    {
      id: '3dcc411f0a3fb9928928be20187503a5',
      absoluteIndex: 62,
      name: 'item-62',
    },
    {
      id: '7d9a0819675274d487199e4a7107bdb2',
      absoluteIndex: 63,
      name: 'item-63',
    },
    {
      id: '5ec9f0a960af5a7acd27bb53e7d26a63',
      absoluteIndex: 64,
      name: 'item-64',
    },
    {
      id: '1ead171c46f482339e281292547bfa84',
      absoluteIndex: 65,
      name: 'item-65',
    },
    {
      id: '05c498bb606a718fbcb36d785e1ec291',
      absoluteIndex: 66,
      name: 'item-66',
    },
    {
      id: '4800f79743bfcb51c4ea1f42257f8ba3',
      absoluteIndex: 67,
      name: 'item-67',
    },
    {
      id: '96a7ba878bfeef87d03e05c111c0d12b',
      absoluteIndex: 68,
      name: 'item-68',
    },
    {
      id: '89386667437bea2f679a2186cdd4e7bb',
      absoluteIndex: 69,
      name: 'item-69',
    },
    {
      id: '95d0cdfddadac847a717c0ff38cd4ce4',
      absoluteIndex: 70,
      name: 'item-70',
    },
    {
      id: 'ce7bfc635c9f7344561910624638c31f',
      absoluteIndex: 71,
      name: 'item-71',
    },
    {
      id: 'd724e32235073c9a0072c8ba63004be9',
      absoluteIndex: 72,
      name: 'item-72',
    },
    {
      id: '28d503b70c0384696501c81b8b51319c',
      absoluteIndex: 73,
      name: 'item-73',
    },
    {
      id: 'e26b3a73f66a16c9d0658f31f3730e75',
      absoluteIndex: 74,
      name: 'item-74',
    },
    {
      id: 'c64156c2ae35edc0e6cf9fed350634ca',
      absoluteIndex: 75,
      name: 'item-75',
    },
    {
      id: '375771667dfabe97ee457fa08a091ba2',
      absoluteIndex: 76,
      name: 'item-76',
    },
    {
      id: '57f42b26ab7da8e9ed5c7211575de6b5',
      absoluteIndex: 77,
      name: 'item-77',
    },
    {
      id: 'a695d04fbf1f4dd9bc26dd8d30881c01',
      absoluteIndex: 78,
      name: 'item-78',
    },
    {
      id: '4463f8b9ce4c352c84a7d9fcfa914161',
      absoluteIndex: 79,
      name: 'item-79',
    },
    {
      id: '0b5b796c67655bad7bbbe2159ef98ccd',
      absoluteIndex: 80,
      name: 'item-80',
    },
    {
      id: 'b77f0ecc1c06036f69adbfd876d71cc4',
      absoluteIndex: 81,
      name: 'item-81',
    },
    {
      id: '9b9ca4295a3a08bbf80e6e39be0efd6a',
      absoluteIndex: 82,
      name: 'item-82',
    },
    {
      id: 'f84ff74f9aa65450fd7662eeaef3f124',
      absoluteIndex: 83,
      name: 'item-83',
    },
    {
      id: '835de0286c96015eb30df333cd4599b8',
      absoluteIndex: 84,
      name: 'item-84',
    },
    {
      id: 'aed9bd38dc12df8fa89d93943f8861bb',
      absoluteIndex: 85,
      name: 'item-85',
    },
    {
      id: '3234479528d9a18432d400797807ccd1',
      absoluteIndex: 86,
      name: 'item-86',
    },
    {
      id: 'c1d9b0b0c15e6ed8842847d8dd161a10',
      absoluteIndex: 87,
      name: 'item-87',
    },
    {
      id: '32ebfe8f16fa3e1f053f64f01340c48a',
      absoluteIndex: 88,
      name: 'item-88',
    },
    {
      id: 'c0fe70233b82c736210f969b6f80be8c',
      absoluteIndex: 89,
      name: 'item-89',
    },
    {
      id: '94d1f02970f4287c738244c7265121c5',
      absoluteIndex: 90,
      name: 'item-90',
    },
    {
      id: 'da6a43255354c352b2bcae58a3e1aee8',
      absoluteIndex: 91,
      name: 'item-91',
    },
    {
      id: '1a6066becde7b346ec4382624feb5fa6',
      absoluteIndex: 92,
      name: 'item-92',
    },
    {
      id: 'a3a5f60fdd5b347ac102d7c2e7bfb151',
      absoluteIndex: 93,
      name: 'item-93',
    },
    {
      id: 'ed35c3723e7ffef36e11b3f19e089115',
      absoluteIndex: 94,
      name: 'item-94',
    },
    {
      id: '5b9a202460659923e82375fc9c1c7ba5',
      absoluteIndex: 95,
      name: 'item-95',
    },
    {
      id: '17452f0cbd3ffe31674b5b112484e209',
      absoluteIndex: 96,
      name: 'item-96',
    },
    {
      id: '7caf4d76e506362e2217a7b4687e0f6a',
      absoluteIndex: 97,
      name: 'item-97',
    },
    {
      id: '1c10ad4f94c2d0423de76d1f0ed6645a',
      absoluteIndex: 98,
      name: 'item-98',
    },
    {
      id: 'd53ceea67d7c6d909db60f98085facbd',
      absoluteIndex: 99,
      name: 'item-99',
    },
  ],
};
