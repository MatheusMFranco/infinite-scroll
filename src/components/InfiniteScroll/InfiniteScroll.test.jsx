import { screen } from '@testing-library/react';
import { renderTheme } from '../../utils/render-theme';
import mock from './mock';
import { InfiniteScroll } from '.';

describe('<InfiniteScroll />', () => {
  beforeEach(() => {
    window.IntersectionObserver = mock;
  });

  it('should render one div', () => {
    expect.assertions(1);
    const fn = jest.fn();
    const { container } = renderTheme(<InfiniteScroll showMore={fn} />);
    expect(container.querySelectorAll('div')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = renderTheme(<InfiniteScroll showMore={fn} />);
    expect(container).toMatchSnapshot();
  });
});
