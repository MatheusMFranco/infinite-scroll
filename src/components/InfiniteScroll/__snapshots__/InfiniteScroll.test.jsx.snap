// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteScroll /> should match snapshot 1`] = `
<div>
  <div />
</div>
`;
